sphinx_gallery/tests/test_full.py

```diff
@@ -87,6 +87,32 @@ def test_timings(sphinx_app):
     assert ('- %s: ' % fname) in status
 
 
+def test_api_usage(sphinx_app):
+    """Test that an api usage page is created."""
+    out_dir = sphinx_app.outdir
+    src_dir = sphinx_app.srcdir
+    # the rst file is empty, all the content is added
+    # in post-processing
+    api_rst = op.join(src_dir, 'sg_api_usage.rst')
+    assert op.isfile(api_rst)
+    # HTML output
+    api_html = op.join(out_dir, 'sg_api_usage.html')
+    assert op.isfile(api_html)
+    with codecs.open(api_html, 'r', 'utf-8') as fid:
+        content = fid.read()
+    # spot check references
+    assert 'href="auto_examples/plot_numpy_matplotlib.html' in content
+    # check used and unused
+    assert 'alt="API unused entries graph"' in content
```

Comment:
The graph only exists if `sphinx.ext.graphviz` is installed and working, right? You probably need to make this part of the test conditional on that being the case. If it's not installed then you should assert this is *not* in the content

Reply:
Ah, that makes sense.